R/printFFTrees_function.R

```diff
@@ -30,8 +38,24 @@ print.FFTrees <- function(x = NULL,
 
   n.cues <- x$trees$definitions$nodes[tree]
 
+  # - Validate data argument: ----
+
+  data <- tolower(data)  # for robustness
+
+  if (!data %in% c("test", "train")){ # unknown value:
+
+    message("The data to print must be 'test' or 'train'. Using 'test' data...")
+    data <- "test"
+  }
 
-  # Intro: ----
+  if (data == "test" & is.null(x$trees$stats$test)){ # use "train" data:
+
+    message("No 'test' data available. Using training data...")
```

Comment:
Similar to my comment above above, I believe that i the user specifies`data = "test"`, and there is no test data, then the function should throw an error (or return nothing). 

We don't want to run the risk of someone thinking they are seeing results from test data, but in reality seeing training data (they'd have to pay close attention to the message here).

Reply:
I modeled this on the similar behavior of `plot.FFTrees()` — see "btw" in previous comment.

Not printing an object at all when it has no test data (and does not explicitly specify `data = "train"`) would seem rather restrictive (as users who do not use test data will be aware of that, but still want to print objects).  Perhaps stepping up to issuing a `warning()` in such cases would be disruptive enough?

Reply:
Oh sorry I misunderstood the functionality before. I understand better now. I am ok with your edit

I just ran some sample code myself, where I fit a tree without test data, then printed the object. I noticed two things

1. The message `No 'test' data available. I'll print the best training tree instead...` is buried in a lot of other messaging (oh gosh I put wayyy too much in when writing `FFTrees()`). So I can imagine most people might not see it.
2. The print method for `FFTrees` is pretty clear about whether the accuracy statistics are for Training or Testing, as it comes up twice in the 'headers'

All that to say I think the `message("No 'test' data available. Using training data...")` is ok, but may not be needed. I'll defer on this one.

```
> heart.fft <- FFTrees(formula = diagnosis ~., 
+                      data = heart.train,
+                      decision.labels = c("Healthy", "Disease"))
Setting goal = 'wacc'
Setting goal.chase = 'wacc'
Setting cost.outcomes = list(hi = 0, mi = 1, fa = 1, cr = 0)
Growing FFTs with ifan:
[==================================================================================================] 100%
Fitting other algorithms for comparison (disable with do.comp = FALSE) ...
> heart.fft
No 'test' data available. I'll print the best training tree instead...
FFTrees 
- Trees: 7 fast-and-frugal trees predicting diagnosis
- Outcome costs: [hi = 0, mi = 1, fa = 1, cr = 0]

FFT #1: Definition
[1] If thal = {rd,fd}, decide Disease.
[2] If cp != {a}, decide Healthy.
[3] If ca > 0, decide Disease, otherwise, decide Healthy.

FFT #1: Training Accuracy
Training data: N = 150, Pos (+) = 66 (44%) 

|          | True + | True - | Totals:
|----------|--------|--------|
| Decide + | hi  54 | fa  18 |      72
| Decide - | mi  12 | cr  66 |      78
|----------|--------|--------|
  Totals:        66       84   N = 150

acc  = 80.0%   ppv  = 75.0%   npv  = 84.6%
bacc = 80.2%   sens = 81.8%   spec = 78.6%

FFT #1: Training Speed, Frugality, and Cost
mcu = 1.74,  pci = 0.87,  E(cost) = 0.200
```

Reply:
I'll switch both statements (in `print.FFTrees()` and `plot.FFTrees()`) from messages to warnings.  As warnings are always displayed below the other output, this renders them more noticeable.  This should satisfice for now, but I'm also open to other solutions in the future, of course.